rs/embedders/benches/embedders_bench/src/lib.rs

```diff
@@ -118,7 +118,12 @@ pub fn update_bench(
                     .unwrap();
                 total_duration += start.elapsed();
                 assert!(matches!(result, WasmResult::Reply(_)));
-                test.checkpoint_canister_memories();
+                match post_setup_action {
+                    PostSetupAction::PerformCheckpoint => {
+                        test.checkpoint_canister_memories();
+                    }
+                    PostSetupAction::None => {}
```

Comment:
Maybe we should change the name from `PostSetupAction` now that it's run every iteration?